libs/ts-rest/core/src/lib/dsl.ts

```diff
@@ -108,6 +110,9 @@ type ApplyOptions<
 > = Omit<TRoute, 'headers'> &
   WithoutUnknown<{
     headers: UniversalMerge<TOptions['baseHeaders'], TRoute['headers']>;
+    strictStatusCodes: TOptions['strictStatusCodes'] extends true
+      ? true
+      : TRoute['strictStatusCodes'];
```

Comment:
I think the route specific option should take precedence regardless of it's true or false. This will allow a user to override the one in the options with whatever they like.
 
```suggestion
    strictStatusCodes: TRoute['strictStatusCodes'] extends boolean
      ? TRoute['strictStatusCodes']
      : TOptions['strictStatusCodes'] extends boolean
      ? TOptions['strictStatusCodes']
      : unknown;
```

The `unknown` type will cause the property to be stripped by the `WithoutUnknown` type.

Reply:
Thanks have made this change!